ethereum/upgrade

```diff
@@ -62,8 +57,9 @@ if [ $ret = 0 ]; then
   exit
 fi
 
-deploy_output=$(npx truffle exec $SCRIPT --network "$truffle_network") || ( echo "✘ $chain $module: $deploy_output" && exit 1 )
-new_implementation=$(echo "$deploy_output" | grep "address:" | cut -d' ' -f3)
+forge script $SCRIPT --rpc-url $rpc_url --private-key $private_key --broadcast --via-ir || ( echo "✘ $chain $module" && exit 1 )
```

Comment:
I don't think `--via-ir` should be set here. We should set the foundry profile appropriately and configure the build pipeline there.

Reply:
Additionally, this makes it hard to use hardware wallets but I'm not sure there's an easy way around this other than editing the script or putting lots of conditionals :thinking:

Reply:
resolved the VIA IR issue!

Reply:
Yeah - you had mentioned this, but in any case this should be a strict improvement over the truffle scripts, right?

Reply:
by the way, is there a github issue mentioning the issue with the forge scripts + hardware wallets you're mentioning? (or, could you describe it again)

Reply:
Hm, I don't know if there is an issue for hardware wallet compatibility. I never opened one at the very least. The point here would be that this particular script uses `--private-key` option here without offering any way to use `--ledger` or `--trezor` for example. I think there are other key store mechanisms supported by foundry too but I personally only care about hardware wallets, at least for now.

Reply:
For a more concrete example you can take a look at how I use hardware wallets here with foundry [here](https://github.com/wormhole-foundation/example-circle-relayer#hardware-wallet-deployment). It's far from ideal because we would probably need to pipe the options through several scripts. I didn't check whether these could be passed as environment variables. :thinking:

Reply:
I see. Does the current upgrade script (using truffle) allow for hardware wallet compatibility? If not, then this PR doesn't add but doesn't remove that functionality either right